addons/point_of_sale/models/pos_config.py

```diff
@@ -585,10 +585,28 @@ def setup_defaults(self, company):
         :param self pos.config: pos.config records present in the company during the installation of localisation.
         :param company res.company: the single company where the pos.config defaults will be setup.
         """
-        self.assign_payment_journals(company)
-        self.generate_pos_journal(company)
-        self.setup_invoice_journal(company)
-
+        if not self.env["pos.config"].search([('company_id', '=', company.id)]):
+            self.create_pos_config_for_company(company)
+        else:
+            self.assign_payment_journals(company)
+            self.generate_pos_journal(company)
+            self.setup_invoice_journal(company)
+
+    def create_pos_config_for_company(self, company):
+        journal = self.env['account.journal'].search([('type', '=', 'purchase'), ('company_id', '=', company.id)], limit=1)
+        invoice_journal = self.env['account.journal'].search([('type', '=', 'sale'), ('company_id', '=', company.id)], limit=1)
+        payment_method = self.env['pos.payment.method'].create({
+                    'name': _('Bank'),
+                    'journal_id': journal.id,
```

Comment:
Same here, the journal is not a journal dedicated to have payment entries. The bank type journal would fit better here.